plugins/inputs/slurm/slurm.go

```diff
@@ -0,0 +1,508 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package slurm
+
+import (
+	"context"
+	stdTls "crypto/tls"
+	_ "embed"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+
+	goslurm "github.com/pcolladosoto/goslurm/v0038"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Slurm struct {
+	URL              string          `toml:"url"`
+	Username         string          `toml:"username"`
+	Token            string          `toml:"token"`
+	IgnoredEndpoints []string        `toml:"ignored_endpoints"`
+	ResponseTimeout  config.Duration `toml:"response_timeout"`
+	tls.ClientConfig
+
+	client      *goslurm.APIClient
+	baseURL     *url.URL
+	endpointMap map[string]bool
+}
+
+func (*Slurm) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *Slurm) Init() error {
+	if s.ResponseTimeout == 0 {
+		s.ResponseTimeout = config.Duration(time.Second * 5)
+	}
+
+	s.endpointMap = map[string]bool{}
+	for _, endpoint := range s.IgnoredEndpoints {
+		s.endpointMap[strings.ToLower(endpoint)] = true
+	}
+
+	if s.URL == "" {
+		return errors.New("empty URL provided")
+	}
+
+	u, err := url.Parse(s.URL)
+	if err != nil {
+		return err
+	}
+
+	if u.Hostname() == "" {
+		return fmt.Errorf("empty hostname for url %q", s.URL)
+	}
+
+	s.baseURL = u
+
+	if u.Scheme != "http" && u.Scheme != "https" {
+		return fmt.Errorf("invalid scheme %q", u.Scheme)
+	}
+
+	var tlsCfg *stdTls.Config
+	if u.Scheme == "https" {
+		tlsCfg, err = s.ClientConfig.TLSConfig()
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Does it make sense to check if a TLS config is given but the scheme is `http` and output a warning? It might be unexpected that the TLS settings are silently ignored in this case...

Reply:
A comment was added to the configuration explaining that if the scheme was `http` TLS options would be silently ignored. However, I do agree it might be better to be explicit about it in the logs which is where I usually look when things are not working as expected myself...

Anyway, I added the warning on b90417b079bad19a68c597dc6fc55ac6531c10cd. Thanks for the suggestion!